### PR TITLE
Fix bug during user publish

### PR DESCRIPTION
**Pull request by srkukarni**
### Motivation

This pr fixes two bugs in the path of user publish
1. Stackoverflow error in ContextImpl where getSchema was calling itself
2. Add topicName while building the producer.

### Modifications

Describe the modifications you've done.

### Result

After your change, what will change.

### CODE REVIEW

File: pulsar-functions/instance/src/main/java/org/apache/pulsar/functions/instance/ContextImpl.java [hunk 1]
```diff
@@ -269,9 +269,11 @@ public ByteBuffer getState(String key) {
 
         if (producer == null) {
             try {
-                Producer<O> newProducer = ((ProducerBuilderImpl<O>) producerBuilder.clone()).schema(schema).create();
+                Producer<O> newProducer = ((ProducerBuilderImpl<O>) producerBuilder.clone())
```
Comment (rdhabalia) [inline pulsar-functions/instance/src/main/java/org/apache/pulsar/functions/instance/ContextImpl.java]: can we please add unit-test here which covers this path.

Comment (srkukarni) [reply pulsar-functions/instance/src/main/java/org/apache/pulsar/functions/instance/ContextImpl.java]: Added

**Comment by srkukarni** [comment]
run integration tests

**Comment by srkukarni** [comment]
run integration tests

**Pull request merged**